### ISSUE TITLE
[bug]: TextInput ref not working as expected

### ISSUE THREAD

**Issue by gmenih** [strapi/design-system#863]
### What version of strapi-design-system are you using?

- Strapi Design system version: 1.6.1

### What's Wrong?

When using `TextInput` element, the `ref` "callback" is called twice - once with the `useImperativeHandle` handler, and once with the `<input>` element. This makes it hard to use.

I'm trying to get it working with `react-hook-form`, but since the ref is called twice, it registers the wrong component.

### To Reproduce

```ts
const SomeComponent = () => {
    const fakeRef = (v) => console.log(v);
    
    return <TextInput label="Some field" ref={fakeRef} />
};
```

This will log the following values:

```
> {inputWrapperRef: HTMLDivElement}
> HTMLInputElement
```

### Expected Behaviour

The ref would only be called once, on the `<input>` element

**Comment by joshuaellis** [comment]
Can you explain a bit more when you say it's hard to use?

IMO id like to remove the imperative handle but this would be a breaking change.

I'm thinking we could spread the current ref (which holds the input node) onto the imperative handle thus preserving its current functionality whilst maintaining "expected" behaviour.

**Comment by INNOVATIVEGAMER** [comment]
> this would be a breaking change.

@joshuaellis Why it is a breaking change ? 
At root level -
1. Why we need the inputWrapperRef for `TextInput` component making an additional wrapper div around it ?
2. The other Input fields like `NumberInput` do not have such wrapper div. So why explicitly we want this to be here ?

**Comment by INNOVATIVEGAMER** [comment]
> I'm thinking we could spread the current ref (which holds the input node) onto the imperative handle thus preserving its current functionality whilst maintaining "expected" behaviour.

Then I also think this will be the way forward. This will enable usage of inputRef directly through imperative handle.
```JS
const SomeComponent = () => {
    const fakeRef = (v) => console.log(v);
    return <TextInput label="Some field" ref={fakeRef} />
};
```
This will log the following - 
`{
inputWrapperRef : HTMLDivElement,
inputRef : HTMLInputElement
}`
@gmenih WDYT ?

**Comment by joshuaellis** [comment]
resolved by [#1693](https://github.com/strapi/design-system/issues/1693)
